### ISSUE TITLE
Attach VisBug to ShadowRoot

### ISSUE THREAD

**Issue by jogibear9988** [GoogleChromeLabs/ProjectVisBug#645]
Idea:

As adding visbug to a page could lead to side effects, wouldn't it be preferable to add it to it's own shadowroot?

Like:

          let sr = document.body.attachShadow({mode: 'open'});
          sr.innerHTML = "<slot></slot>"
        
And after this the VisBug overlays could be added to the ShadowRoot, without affecting the real Dom?

**Comment by argyleink** [comment]
i'm willing to try out a PR and test it 👍🏻 
but visbug does apply attributes to nodes and does some other things that are still "side effects", but reducing them is indeed generally nice

**Comment by jogibear9988** [comment]
I'll look to create a demo

**Comment by jogibear9988** [comment]
> i'm willing to try out a PR and test it 👍🏻 but visbug does apply attributes to nodes and does some other things that are still "side effects", but reducing them is indeed generally nice

I saw that it add's attributes for selection. Is there a reason for this? Why is the selection not stored in some varaibles?

**Comment by jogibear9988** [comment]
Yeah, maybe it's my thinking cause I'm working on a designer, wich exports the html and so my output should not contain own added attributes.
